sdk/core/core-lro/src/azureAsyncOperationStrategy.ts

```diff
@@ -0,0 +1,182 @@
+      }
+
+      if (initialOperationResult.location) {
+        switch (finalStateVia) {
+          case "original-uri":
+            currentOperation = await sendFinalGet(initialOperation, sendOperationFn);
+            return currentOperation;
+
+          case "azure-async-operation":
+            return currentOperation;
+          case "location":
+          default:
+            // eslint-disable-next-line no-case-declarations
+            const location = initialOperationResult.location || currentOperationResult.location;
+
+            if (!location) {
+              throw new Error("Couldn't determine final GET URL from location");
+            }
+
+            // eslint-disable-next-line no-case-declarations
+            const result: LROOperationStep<TResult> = await sendFinalGet(
+              initialOperation,
+              sendOperationFn,
+              location
+            );
+            return result;
+        }
+      }
+
+      // All other cases return the last operation
+      return currentOperation;
+    },
+    poll: async () => {
+      if (!lastKnownPollingUrl) {
+        throw new Error("Unable to determine polling url");
+      }
+
+      const pollingArgs = currentOperation.args;
+      // Make sure we don't send any body to the get request
+      const { requestBody, ...restSpec } = currentOperation.spec;
+      const pollingSpec: OperationSpec = {
+        ...restSpec,
+        httpMethod: "GET",
+        path: lastKnownPollingUrl
+      };
+
+      const result = await sendOperationFn(pollingArgs, pollingSpec);
+
+      // Update latest polling url
+      lastKnownPollingUrl =
+        result._lroData?.azureAsyncOperation ||
+        result._lroData?.operationLocation ||
+        lastKnownPollingUrl;
+
+      // Update lastOperation result
+      currentOperation = {
+        args: pollingArgs,
+        spec: pollingSpec,
+        result
+      };
+
+      return currentOperation;
+    }
+  };
+}
+
+function shouldPerformFinalGet(
+  initialResult: LROResponseInfo,
+  currentResult: LROResponseInfo
+): boolean {
+  const { status } = currentResult;
+  const { requestMethod: initialRequestMethod, location } = initialResult;
+  if (status && status.toLowerCase() !== "succeeded") {
+    return false;
+  }
+
+  if (initialRequestMethod === "DELETE") {
+    return false;
+  }
+
+  if (initialRequestMethod !== "PUT" && !location) {
+    return false;
+  }
+
+  return true;
+}
+
+async function sendFinalGet<TResult extends BaseResult>(
+  initialOperation: LROOperationStep<TResult>,
+  sendOperationFn: SendOperationFn<TResult>,
+  path?: string
+): Promise<LROOperationStep<TResult>> {
+  // Make sure we don't send any body to the get request
+  const { requestBody, ...restSpec } = initialOperation.spec;
+  const finalGetSpec: OperationSpec = {
+    ...restSpec,
+    httpMethod: "GET"
+  };
+
+  // Send final GET request to the Original URL
+  const spec = {
+    ...finalGetSpec,
+    ...(path && { path })
```

Comment:
Is the idea here you only want `path` on the `spec` object when it exists? I feel like it could just be

```suggestion
    path
```

and either it'll be undefined or not